src/uct/api/uct.h

```diff
@@ -1238,7 +1238,7 @@ struct uct_md_attr {
         uint64_t             reg_mem_types; /**< Bitmap of memory types that Memory Domain can be registered with */
         uint64_t             detect_mem_types; /**< Bitmap of memory types that Memory Domain can detect if address belongs to it */
         uint64_t             alloc_mem_types;  /**< Bitmap of memory types that Memory Domain can allocate memory on */
-        ucs_memory_type_t    access_mem_type; /**< Memory type that Memory Domain can access */
+        uint64_t             access_mem_types; /**< Memory types that Memory Domain can access */
```

Comment:
@shamisp sorry I missed your messages. 

> Do we have the cases when alloc_mem_types != access_mem_types ?

This is true for IB transport at least. It can access cuda, host, etc but cannot allocate any type.

> ```This structure defines the attributes of a Memory Domain which includes maximum memory that can be allocated, credentials required for accessing the memory, maximum memory that can be allocated, credentials required for accessing the memory, CPU mask indicating the proximity of CPUs, and bitmaps indicating the types of memory (CPU/CUDA/ROCM) that can be detected, allocated and accessed```

Agree with the last addition but I think `credentials required for accessing the memory` has been repeated. I'll remove that.